yt_dlp/extractor/tubetugraz.py

```diff
@@ -0,0 +1,349 @@
+# coding: utf-8
+from __future__ import unicode_literals
+from .common import InfoExtractor
+from ..utils import ExtractorError, urlencode_postdata, try_get
+from collections import defaultdict
+import re
+
+
+def try_get_or(src, getters, expected_type=None):
+    for getter in getters:
+        v = try_get(src, getter, expected_type)
+        if v is not None:
+            return v
+
+
+class TubeTuGrazIE(InfoExtractor):
+    IE_DESC = 'tube.tugraz.at'
+    IE_NAME = 'TubeTuGraz'
+
+    _VALID_URL = r"""(?x)^
+        (?:https?://)?tube\.tugraz\.at/paella/ui/(?:
+            (?P<series>browse\.html\?series=)|
+            (?P<episode>watch.html\?id=)
+        )(?P<id>[0-9a-fA-F]{8}-(?:[0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12})
+    $"""
+    _TEST = {
+        "url": "https://tube.tugraz.at/paella/ui/watch.html?id=f2634392-e40e-4ac7-9ddc-47764aa23d40",
+        "md5": "a23a3d5c9aaca2b84932fdba66e17145",
+        "info_dict": {
+            "id": "f2634392-e40e-4ac7-9ddc-47764aa23d40",
+            "ext": "mp4",
+            "title": "#6 (23.11.2017)",
+            "episode": "#6 (23.11.2017)",
+            "series": "[INB03001UF] Einführung in die strukturierte Programmierung",
+            "creator": "Safran C",
+            "duration": 3295818,
+        }
+    }
+
+    _LOGIN_REQUIRED = False
+    _NETRC_MACHINE = None
+
+    _LOGIN_URL = "https://tube.tugraz.at/Shibboleth.sso/Login?target=/paella/ui/index.html"
+    _LOGIN_SUCCESS_URL = "https://tube.tugraz.at/paella/ui/index.html"
+
+    _API_SERIES = "https://tube.tugraz.at/series/series.json"
+    _API_EPISODE = "https://tube.tugraz.at/search/episode.json"
+
+    def _login(self):
+        username, password = self._get_login_info()
+        if username is None:
+            return
+
+        result = self._download_webpage_handle(
```

Comment:
_request_webpage

Reply:
done for this request, don't know exactly how to do that for the first request